azavea/raster-foundry
# Add extent field and computation to project

## Overview

Computes new extent on scene addition or removal from project. This currently leans on GeoTrellis to compute the extent. Moving this computation to the DB may be an optimization to look at in the future.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~

## Notes

This does not change the front-end to use the new extent field.

## Testing Instructions

 * Open sbt console (`./scripts/console app-server ./sbt`)
 * Ensure migrations are run (`mg update`, `mg apply`)
 * Run tests
 * Exit console, start app-server (`./scripts/server`)
 * Start front-end
 * Open app and dev tools to look at requests
 * Create a new project, do not add any scenes
 * Request the project in some way, see that the extent field isn't returned (this is expected behavior)
 * Add a scene to the project
 * Request the project again, see that the extent field is returned and is correct (compare with scene, drop into geojson.io or the like)
 * Remove the scene from the project
 * Ensure that the extent is removed from the project
 * Do various combinations of adding and removing scenes from the project, see that the extents are correct for that project

Connects #999
